spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/core/PubSubTemplate.java

```diff
@@ -124,4 +160,11 @@ public void setMessageConverter(MessageConverter messageConverter) {
 		this.messageConverter = messageConverter;
 	}
 
+	public int getExecutorThreadCount() {
+		return this.executorThreadCount;
+	}
+
+	public void setExecutorThreadCount(int executorThreadCount) {
+		this.executorThreadCount = executorThreadCount;
```

Comment:
??? That doesn't fit our Spring goal. We provide here some high level lightweight wrapping API to fit it into dependency injection idioms. But if there is some helpers API (builders ?) which can be used directly as beans and fully feasible in the inversion of control container that would be overhead to wrap that API to something else.
Right, if we look into the XML configuration, where builder API isn't executable, we need to wrap it into `FactoryBean` implementation. But here it is fully fine to let end-user to deal with the `InstantiatingExecutorProvider.newBuilder()` directly and inject the result into this `PubSubTemplate`. Having that I don't see reason to overhead the `PubSubTemplate` configuration with wider option and restricted one. Therefore my vote only for the `executorProvider` option.